src/main/java/services/DatafeedEventsServiceV2.java

```diff
@@ -0,0 +1,132 @@
+package services;
+
+import clients.SymBotClient;
+import exceptions.APIClientErrorException;
+import lombok.SneakyThrows;
+import model.DatafeedEvent;
+import model.datafeed.DatafeedV2;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * DatafeedEventsServiceV2 providing services to bot for subscribing the datafeed version 2.
+ */
+class DatafeedEventsServiceV2 extends AbstractDatafeedEventsService {
+
+    private static final Logger logger = LoggerFactory.getLogger(DatafeedEventsServiceV2.class);
+
+    private static final int AWAIT_TERMINATION = 1000;
+
+    private final AtomicBoolean started = new AtomicBoolean(false);
+    private ExecutorService threadPool;
+
+    public DatafeedEventsServiceV2(SymBotClient client) {
+        super(client);
+        this.readDatafeed();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void readDatafeed() {
+        if (threadPool != null) {
+            threadPool.shutdownNow();
+        }
+        threadPool = Executors.newSingleThreadExecutor();
+        started.set(true);
+        threadPool.submit(() -> {
+            do {
+                try {
+                    readEventsFromDatafeed();
+                } catch (APIClientErrorException e) {
+                    //Datafeed was stale
+                    logger.info(e.getMessage());
+                } catch (Exception e) {
+                    logger.error("Something went wrong while reading datafeed", e);
+                    logger.info("Sleeping for {} seconds before retrying..", botClient.getConfig().getDatafeedEventsErrorTimeout());
+                    sleep(botClient.getConfig().getDatafeedEventsErrorTimeout());
+                }
+            } while (started.get());
+        });
+    }
+
+    @SneakyThrows
+    private void readEventsFromDatafeed() {
+        String datafeedId;
+        List<DatafeedV2> datafeedIds = datafeedClient.listDatafeedId();
+        if (datafeedIds.isEmpty()) {
+            datafeedId = this.createDatafeedId();
+        } else {
+            //Each bot should subscribe only one datafeed
+            datafeedId = datafeedIds.get(0).getId();
+        }
+        //Read datafeed in loop
+        do {
+            List<DatafeedEvent> events = datafeedClient.readDatafeed(datafeedId, datafeedClient.getAckId());
+            if (events != null && !events.isEmpty()) {
+                handleEvents(events);
+            }
+
+        } while (started.get());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @SneakyThrows
```

Comment:
do we still need it?
from lombok's doc _This somewhat contentious ability should be used carefully, of course_
I would probably avoid using it for public APIs and be explicit on which exceptions can be raised so a user knows which error cases it has to handle